blog/views.py

```diff
@@ -0,0 +1,58 @@
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.views import generic
+from django.views.generic.edit import FormMixin
+from .forms import CommentaryForm
+
+from blog.models import Post
+
+
+class IndexView(generic.ListView):
+    model = Post
+    posts = Post.objects.all().order_by("-created_time")
+    paginate_by = 5
+    template_name = "blog/index.html"
+    queryset = Post.objects.all().prefetch_related("commentary")
+
+
+class PostList(generic.ListView):
+    model = Post
+    template_name = "blog/post_list.html"
+    context_object_name = "post_list"
+    queryset = Post.objects.all().prefetch_related(
+        "commentary__user", "commentary__user__posts"
+    )
```

Comment:
The 'queryset' attribute in PostList is correctly using 'prefetch_related' for optimizing database queries. Ensure that 'commentary__user__posts' is necessary for your use case, as it might add unnecessary complexity if not needed.